### DISCUSSION TITLE
How can I reach LocalAi from another network machine?

### DISCUSSION THREAD

**Discussion by griffindodd** [mudler/LocalAI#1780] [Q&A]
I can get correct CURL responses from both within the localAi docker instance and from the docker host machine at localhost:11438, however when I try and hit the host machine's <IP>:11438 from another network machine I get the error 'failed to connect to localhost port 8080' in the curl response.

I don't have any firewall rules running on the host.

is there a setting I have to change to allow the backend to speak to machines within my network.

**Comment by etlweather** [comment]
Linux? Windows with WSL2? What is your `docker` command?

**Reply by etlweather** [reply]
Also give output from console when you start LocalAI. Looks like this:

```
 ┌───────────────────────────────────────────────────┐
 │                   Fiber v2.50.0                   │
 │               http://127.0.0.1:8080               │
 │       (bound on host 0.0.0.0 and port 8080)       │
 │                                                   │
 │ Handlers ........... 110  Processes ........... 1 │
 │ Prefork ....... Disabled  PID ................ 14 │
 └───────────────────────────────────────────────────┘
```

**Comment by JackBekket** [comment]
> I can get correct CURL responses from both within the localAi docker instance and from the docker host machine at localhost:11438, however when I try and hit the host machine's :11438 from another network machine I get the error 'failed to connect to localhost port 8080' in the curl response.
> 
> I don't have any firewall rules running on the host.
> 
> is there a setting I have to change to allow the backend to speak to machines within my network.

localhost is yourself, when you try to connect to localhost -- means you call yourself at port 11438 (by default it should be 8080, as web-server)

when you try to call localhost:8080 -- you are not calling to localai machine, you are calling yourself. 

basically you should replace localhost with actual address of machine with localai and try it

it may be also tricky with ports
